src/prog_models/sim_result.py

```diff
@@ -44,6 +46,8 @@ def __getitem__(self, item):
     
     def __iter__(self):
         """
+        .. versionadded:: 1.5.0
+        
```

Comment:
```suggestion
```
Implemented it in this release to add deprecation warning only- We don't need the message for it